src/esw/network_monitor.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+import time
+import rospy
+from mrover.msg import NetworkBandwidth
+
+# Credit: https://stackoverflow.com/questions/15616378/python-network-bandwidth-monitor
+
+
+def get_bytes(t: str, interface: str) -> int:
+    with open("/sys/class/net/" + interface + "/statistics/" + t + "_bytes", "r") as f:
+        data = f.read()
+        return int(data)
+
+
+if __name__ == "__main__":
+
+    rospy.init_node("network_monitor")
+
+    iface = rospy.get_param("network_iface")
+
+    pub = rospy.Publisher("network_bandwidth", NetworkBandwidth, queue_size=1)
+
+    while True:
+        tx1 = get_bytes("tx", iface)
+        rx1 = get_bytes("rx", iface)
+
+        time.sleep(1)
+
+        tx2 = get_bytes("tx", iface)
+        rx2 = get_bytes("rx", iface)
+
+        tx_speed = (tx2 - tx1) / 1000000.0  # MBps
+        rx_speed = (tx2 - tx1) / 1000000.0  # MBps
```

Comment:
For starters I think this is supposed to say rx2 - rx1.

Also is there any risk of overflow with this if we leave it running for a while? Maybe we should add a check for if the second data point is less than the first and not publish if that's the case?

Reply:
There risk of over flow is so small, that there is basically no concern. Also in python ints are promoted to longs. Also, even if there was a cap at 64 bits for a long, even if we somehow used 1 GBps (2^30 Bps), it would around 2^61 / 2^30 seconds or 68 years for it to overflow